include/type.hpp

```diff
@@ -205,6 +211,10 @@ class StructType : public RecordType {
 
   std::string id()  // NOLINT(readability-identifier-naming)
       const noexcept override;
+  std::size_t offset(  // NOLINT(readability-identifier-naming)
+      const std::string& id) const override;
+  std::size_t offset(  // NOLINT(readability-identifier-naming)
+      const std::size_t index) const override;
```

Comment:
I suggest dropping the `const` specifier for pass-by-value parameters from the function declaration, but keep it in the function definition. This does not apply to `const`-reference parameters, which should be kept in both places.

```suggestion
  std::size_t offset(  // NOLINT(readability-identifier-naming)
      std::size_t index) const override;
```

Note that two function declarations differing only in the constness of their pass-by-value parameters are not considered overloads, as they are indistinguishable when called. Removing `const` from the function declaration is recommended for the following reasons:

1. The caller does not need to know whether the parameter is `const` or not, as it does not affect the caller. Including `const` only clutters the function declaration.
2. You might want to change the function's implementation, which could involve removing the `const`. If `const` is included in the function declaration, you will have to update it there as well.